source/vibe/core/process.d

```diff
@@ -0,0 +1,758 @@
+    const string[string] env = null,
+    Config config = Config.none,
+    scope NativePath workDir = NativePath.init)
+@trusted {
+    return Process(eventDriver.processes.spawn(
+        args,
+        ProcessStdinFile(ProcessRedirect.inherit),
+        ProcessStdoutFile(ProcessRedirect.inherit),
+        ProcessStderrFile(ProcessRedirect.inherit),
+        env,
+        config,
+        workDir.toNativeString()).pid
+    );
+}
+
+/// ditto
+Process spawnProcess(
+    scope string program,
+    const string[string] env = null,
+    Config config = Config.none,
+    scope NativePath workDir = NativePath.init)
+{
+    return spawnProcess(
+        [program],
+        env,
+        config,
+        workDir
+    );
+}
+
+/// ditto
+Process spawnShell(
+    scope string command,
+    const string[string] env = null,
+    Config config = Config.none,
+    scope NativePath workDir = NativePath.init,
+    scope NativePath shellPath = nativeShell)
+{
+    return spawnProcess(
+        shellCommand(command, shellPath),
+        env,
+        config,
+        workDir);
+}
+
+private string[] shellCommand(string command, NativePath shellPath)
+{
+    version (Windows)
+    {
+        // CMD does not parse its arguments like other programs.
+        // It does not use CommandLineToArgvW.
+        // Instead, it treats the first and last quote specially.
+        // See CMD.EXE /? for details.
+        return [
+            std.process.escapeShellFileName(shellPath.toNativeString())
+                ~ ` /C "` ~ command ~ `"`
+        ];
+    }
+    else version (Posix)
+    {
+        return [
+            shellPath.toNativeString(),
+            "-c",
+            command,
+        ];
+    }
+}
+
+/**
+    Represents a running process.
+*/
+struct Process {
+    private static struct Context {
+        //Duration waitTimeout;
+        shared(NativeEventDriver) driver;
+    }
+
+    private {
+        ProcessID m_pid;
+        Context* m_context;
+    }
+
+    private this(ProcessID p)
+    nothrow {
+        assert(p != ProcessID.invalid);
+        m_pid = p;
+        m_context = () @trusted { return &eventDriver.processes.userData!Context(p); } ();
+        m_context.driver = () @trusted { return cast(shared)eventDriver; } ();
+    }
+
+    this(this)
+    nothrow {
+        if (m_pid != ProcessID.invalid)
+            eventDriver.processes.addRef(m_pid);
+    }
+
+    ~this()
+    nothrow {
+        if (m_pid != ProcessID.invalid)
+            releaseHandle!"processes"(m_pid, m_context.driver);
+    }
+
+    /**
+        Check whether this is a valid process handle. The process may have
+        exited already.
+    */
+    bool opCast(T)() const nothrow if (is(T == bool)) { return m_pid != ProcessID.invalid; }
+
+    ///
+    unittest {
+        Process p;
+
+        assert(!p);
+    }
+
+    /**
+        An operating system handle to the process.
+    */
+    @property int osHandle() const nothrow @nogc { return cast(int)m_pid; }
```

Comment:
If there is nothing that speaks against it, I'd suggest to simply stick to `pid` as the name, as handle is slightly confusing (Windows uses `HANDLE` to access processes, in addition to the usual PID).

Reply:
Ah, I wasn't aware windows had both PID and a HANDLE. `pid` makes a lot more sense then :)